bl31/bl31.mk

```diff
@@ -12,6 +12,7 @@ BL31_SOURCES		+=	bl31/bl31_main.c				\
 				bl31/aarch64/runtime_exceptions.S		\
 				bl31/aarch64/crash_reporting.S			\
 				bl31/bl31_context_mgmt.c			\
+				bl31/plat_params.c				\
```

Comment:
I think the mechanism needs to be flexible as to whether the config is parsed before or after its passed to each BL. MMU enablement is separate issue; in future, it might be possible to keep the MMU enabled when passing from one BL to the next, but that requires changes to the handover interface and is not something we're currently looking at.

I want to find a solution that works for you, as well as for partners who want a more traditional device tree approach.